domain/storage/state/storagepool.go

```diff
@@ -501,10 +521,22 @@ WHERE  sp.name = $storagePool.name`, inputArg, poolAttribute{})
 	}
 
 	if len(storagePools) == 0 {
-		return domainstorage.StoragePool{}, errors.Errorf("storage pool %q", name).Add(storageerrors.PoolNotFoundError)
+		return domainstorage.StoragePool{}, errors.Errorf("storage pool %q", uuid).Add(storageerrors.PoolNotFoundError)
 	}
 	if len(storagePools) > 1 {
 		return domainstorage.StoragePool{}, errors.Errorf("expected 1 storage pool, got %d", len(storagePools))
 	}
 	return storagePools[0], errors.Capture(err)
 }
+
+// GetStoragePoolByName returns the storage pool with the specified name.
+// The following errors can be expected:
+// - [storageerrors.PoolNotFoundError] if a pool with the specified name does not exist.
+// Exported for use by other domains that need to load storage pools.
+func GetStoragePoolByName(ctx context.Context, db domain.TxnRunner, name string) (domainstorage.StoragePool, error) {
```

Comment:
Do we need this version?

Reply:
Yeah, it's used by the application domain https://github.com/ycliuhw/juju/blob/36e80a3944671826e8db0a60b294f11cb6eb06dc/domain/application/state/application.go#L668.
Probably we should change the app domain implementation to get by UUID as well.

Reply:
I think what @tlm is saying is: Service code can call a `GetStoragePoolUUID` then call `GetStoragePool`. Since UUIDs are unique, we only incur the cost of an extra read transaction. The outcome is, most state methods
just need to understand uuid, then the service layer can offer both if both are needed.